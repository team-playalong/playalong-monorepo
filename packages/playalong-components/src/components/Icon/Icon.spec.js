import * as React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import PlyIcon from './Icon';

describe('PlyIcon Component', () => {
  let wrapper;
  const props = {
    icon: 'test.svg',
    click: jest.fn(),
    tooltip: 'Hello There',
  };



  beforeEach(() => {
    wrapper = shallow(<PlyIcon {...props} />);
  });

  test('Should always be true', () => {
    expect(true).toBe(true);
  });

  test('BtnIcon should have an icon property', () => {
    expect(wrapper.childAt(0).props().name).toBe(props.icon);
  });

  test('BtnIcon should respond to click events', () => {
    const ev = {};
    wrapper.simulate('click', ev);
    expect(props.click.mock.calls[0][0]).toBe(ev);
  });

  test('BtnIcon have a tooltip', () => {
    expect(wrapper.find('[data-tip]').props()['data-tip']).toBe(props.tooltip);
  });

  test('Icon should support flags', () => {
    props.flag = 'il';
    wrapper = shallow(<PlyIcon {...props} />).childAt(0);

    expect(wrapper.hasClass('flag-icon-il')).toBe(true);
  });

  test('Snapshot testing', () => {
    const tree = renderer.create(<PlyIcon {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
