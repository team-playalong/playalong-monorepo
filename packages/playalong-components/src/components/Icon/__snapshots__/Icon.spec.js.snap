// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlyIcon Component Snapshot testing 1`] = `
<span
  data-tip="Hello There"
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
      "fontSize": "initial",
      "padding": "5px",
      "verticalAlign": "middle",
    }
  }
>
  <span
    className="flag-icon flag-icon-il"
  />
  <div
    className="__react_component_tooltip place-top type-dark "
    data-id="tooltip"
  >
    
  </div>
</span>
`;
